This is BCM4709 (AKA BCM4709A0) based router with ports-on-the-front
board design.

---
 arch/arm/boot/dts/Makefile                           |  1 +
 arch/arm/boot/dts/bcm4709-xiaomi-r1d.dts             | 63 +++++++++++++++++++++++++++
 2 files changed, 64 insertions(+)
 create mode 100644 arch/arm/boot/dts/bcm4709-xiaomi-r1d.dts

--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -98,0 +98,1 @@ dtb-$(CONFIG_ARCH_BCM_5301X) += \
+ 	bcm4709-xiaomi-r1d.dtb \
--- /dev/null
+++ b/arch/arm/boot/dts/bcm4709-xiaomi-r1d.dts
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2021 Xiaomi.
+ *
+ * Licensed under the ISC license.
+ */
+
+/dts-v1/;
+
+#include "bcm4709.dtsi"
+
+/ {
+	compatible = "xiaomi,r1d", "brcm,bcm4709", "brcm,bcm4708";
+	model = "XIAOMI R1D";
+
+	chosen {
+		bootargs = "console=ttyS0,115200 earlycon";
+	};
+
+	memory {
+		reg = <0x00000000 0x08000000
+		       0x88000000 0x08000000>;
+	};
+};
+
+&usb2 {
+	vcc-gpio = <&chipcommon 13 GPIO_ACTIVE_HIGH>;
+};
+
+&spi_nor {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot@0 {
+			label = "boot";
+			reg = <0x000000 0x040000>;
+			read-only;
+		};
+
+		linux@40000 {
+			label = "linux";
+			reg = <0x40000 0x200000>;
+		};
+
+		rootfs@240000 {
+			label = "rootfs";
+			reg = <0x240000 0xda0000>;
+		};
+
+		board_data@fe0000 {
+			label = "board_data";
+			reg = <0xfe0000 0x10000>;
+		};
+
+		nvram@ff0000 {
+			label = "nvram";
+			reg = <0xff0000 0x010000>;
+		};
+	};
+};
